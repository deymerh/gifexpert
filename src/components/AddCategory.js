import React, { useState } from 'react';
import propTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const handleChange = (e) => {
    setInputValue(e.target.value)
  }
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 3) {
      setCategories(cats => [inputValue, ...cats]);
      setInputValue('')
    }
  }
  return (
    <form onSubmit={handleSubmit}>
      <input
        value={inputValue}
        onChange={handleChange}
        type="text" />
    </form>
  )
}
AddCategory.propTypes = {
  setCategories: propTypes.func.isRequired
}