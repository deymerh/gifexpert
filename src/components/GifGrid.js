import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const { loading, data: images } = useFetchGifs(category);

  return loading ? <p>Cargando...</p> : (
    <>
      <h3> {category} </h3>
      <div className="card-grid animate__animated animate__fadeIn">
        {
          images.map(img => {
            return (
              <GifGridItem key={img.id} {...img} />
            )
          })
        }
      </div>
    </>
  )
}
