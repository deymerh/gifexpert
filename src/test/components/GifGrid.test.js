import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import React from 'react';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el componentes <GifGrid />', () => {

  const category = 'One Punch';

  test('Debe mostrarse correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar items cuando se cargan imagenes de mi custom hook useFetchGifs', () => {
    const gifs = [
      {
        id: '123',
        url: 'https://localhost/cualquiera/imagen.jpg',
        title: 'Lo que sea'
      },
      {
        id: '123',
        url: 'https://localhost/cualquiera/imagen.jpg',
        title: 'Lo que sea'
      }
    ]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  })


})
