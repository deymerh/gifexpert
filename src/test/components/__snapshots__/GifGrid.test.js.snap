// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componentes <GifGrid /> Debe mostrar items cuando se cargan imagenes de mi custom hook useFetchGifs 1`] = `
<Fragment>
  <h3>
     
    One Punch
     
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <GifGridItem
      id="123"
      key="123"
      title="Lo que sea"
      url="https://localhost/cualquiera/imagen.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Lo que sea"
      url="https://localhost/cualquiera/imagen.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componentes <GifGrid /> Debe mostrarse correctamente 1`] = `
<p>
  Cargando...
</p>
`;
